src/main/scala/microsites/MicrositeKeys.scala

```diff
@@ -551,6 +557,25 @@ trait MicrositeAutoImportSettings extends MicrositeKeys {
       extracted.runTask(publishMultiversionMicrosite, st)._1
     }
 
+  protected[this] def buildGithubConfig(hostingUrl: String)(implicit log: Logger): GithubConfig =
+    Either.catchOnly[MalformedURLException](new URL(hostingUrl)) match {
+      case Right(url) if url.getHost == "github.com" => GithubConfig.default
+      case Right(url) if url.getProtocol.startsWith("https") =>
+        val replaceHost: String => String = s => s.replace("github.com", url.getHost)
+        GithubConfig.default
+          .copy(
+            baseUrl = s"${url.getProtocol}://${url.getHost}/api/v3/",
```

Comment:
This is my only concern, is this always `v3`?

Reply:
The research I did, REST API is only on v3 (and there is a new GraphQL API on v4)
https://github.com/orgs/community/discussions/24559
I didn't see any GitHub Enterprise mention of v2 or v4 so I didn't go ahead with soft-coding this value.
But if you think we should, I can make that change.

Reply:
To the GitHub Enterprise Server versions, for Cloud version it changes to` https://api.github.com`
https://docs.github.com/en/enterprise-server@3.4/rest/quickstart
https://docs.github.com/en/enterprise-cloud@latest/rest/quickstart?apiVersion=2022-11-28

Reply:
Could you please add an issue with the information? Thanks

Reply:
I hope I'm not adding confusion, but there is no issue. When it's a GH Enterprise target it's `/api/v3` , when it's GitHub, it's `api.github.com`.
This PR handles both cases, defaulting to https://github.com/47degrees/github4s/blob/main/github4s/shared/src/main/scala/github4s/GithubConfig.scala#L43 when hostname is `github.com`.